pkg/events/events.go

```diff
@@ -164,6 +195,31 @@ func New(ctx context.Context, name string, identifiers CombinedIdentifiers, data
 	if identifiers != nil {
 		evt.innerEvent.Identifiers = identifiers.CombinedIdentifiers().GetEntityIdentifiers()
 	}
+	authentication := &ttnpb.Event_Authentication{}
+	if p, ok := peer.FromContext(ctx); ok && p.Addr.String() != "pipe" {
+		host, _, err := net.SplitHostPort(p.Addr.String())
+		if err == nil {
+			authentication.RemoteIP = host
+		}
+	}
+	md := rpcmetadata.FromIncomingContext(ctx)
+	if md.AuthType != "" {
+		authentication.Type = md.AuthType
+	}
+	if md.AuthValue != "" {
+		// Cannot use auth.SplitToken() here because of import cycle
```

Comment:
Then we should probably fix that differently. `pkg/auth` itself isn't importing `pkg/events`. Perhaps the cycle is because `password_test.go` uses `package auth` instead of `package auth_test`, which should be an easy fix.

Reply:
Actually it is `auth_test` that imports `assertions/should`, which in turn imports `pkg/events`. We cannot avoid that.